### ISSUE TITLE
Some examples crash on minimise

### ISSUE THREAD

**Issue by crispweed** [gfx-rs/wgpu-rs#531]
I get an error something like this:

```
     Running `target\debug\examples\skybox.exe`
thread 'main' panicked at 'The aspect ratio cannot be below zero, found: NaN', /rustc/d3fb005a39e62501b8b0b356166e515ae24e2e54\src\libstd\macros.rs:16:9
```

This happens on each of the examples that changes projection parameters on resize, (as opposed to just scaling the rendered view) apart from the water example.

I have a desktop split across two screens, on Windows, and don't know if the issue repeats on other setups, but changing the resize() function in the cube example to the following fixes the issue for me:

```
fn resize(
        &mut self,
        sc_desc: &wgpu::SwapChainDescriptor,
        _device: &wgpu::Device,
        queue: &wgpu::Queue,
    ) {
        let safeWidth = if sc_desc.width > 0 { sc_desc.width } else { 1 };
        let safeHeight = if sc_desc.height > 0 { sc_desc.height } else { 1 };
        let mx_total = Self::generate_matrix(safeWidth as f32 / safeHeight as f32);
        let mx_ref: &[f32; 16] = mx_total.as_ref();
        queue.write_buffer(&self.uniform_buf, 0, bytemuck::cast_slice(mx_ref));
    }
```

**Comment by kvark** [comment]
If we can fix it in the example framework, that's wonderful! Please make a PR ❤️

**Comment by crispweed** [comment]
Hmm.. this one maybe shouldn't be closed yet.
So yeah, we can fix this in the example framework instead of in each individual example, as you say, and there's a commit on my fork that does this (that we can see referencing this issue, above), but I'm not so hot on git and haven't been able to work out how to make a pull request for just this one commit by itself.. :/

**Comment by crispweed** [comment]
Actually it looks like the fix for this issue *has* been merged in, together with the fix for #532, somehow, and then this issue *should* now be closed. (Sorry for the noise!)
